packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -71,19 +117,93 @@ class TimeInput extends React.Component {
     this.handleFocus = this.handleFocus.bind(this);
     this.handleHourBlur = this.handleHourBlur.bind(this);
     this.handleMinuteBlur = this.handleMinuteBlur.bind(this);
+    this.handleMeridiemChange = this.handleMeridiemChange.bind(this);
+    this.handleMeridiemInputKeyDown = this.handleMeridiemInputKeyDown.bind(this);
+    this.meridiemFocus = this.meridiemFocus.bind(this);
+    this.meridiemBlur = this.meridiemBlur.bind(this);
+    this.handleMeridiemButtonChange = this.handleMeridiemButtonChange.bind(this);
+
+    let hour = TimeUtil.splitHour(value);
+    let meridiem = null;
+
+    if (props.variant === FORMAT_12_HOUR) {
+      this.anteMeridiem = context.intl.formatMessage({ id: 'Terra.timeInput.am' });
+      this.postMeridiem = context.intl.formatMessage({ id: 'Terra.timeInput.pm' });
+
+      if (hour) {
+        const parsedHour = this.parseTimeTwelveHourTime(hour, true);
+        hour = parsedHour.hourString;
+        meridiem = parsedHour.meridiem;
+      } else {
+        meridiem = this.anteMeridiem;
+      }
+    }
+
+    this.state = {
+      hour,
+      minute: TimeUtil.splitMinute(value),
+      isFocused: false,
+      meridiem,
+      meridiemFocused: false,
+    };
   }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.value === this.props.value) {
+    if (
+      nextProps.value === this.props.value &&
+      nextProps.variant === this.props.variant
+    ) {
       return;
     }
 
+    let hour = TimeUtil.splitHour(nextProps.value);
+    let meridiem = this.state.meridiem;
+
+    if (nextProps.variant === FORMAT_12_HOUR) {
+      this.anteMeridiem = this.context.intl.formatMessage({ id: 'Terra.timeInput.am' });
+      this.postMeridiem = this.context.intl.formatMessage({ id: 'Terra.timeInput.pm' });
+
+      if (hour) {
+        const parsedHour = this.parseTimeTwelveHourTime(hour, false);
+        hour = parsedHour.hourString;
+        meridiem = parsedHour.meridiem;
+      }
+    }
+
     this.setState({
-      hour: TimeUtil.splitHour(nextProps.value),
+      hour,
       minute: TimeUtil.splitMinute(nextProps.value),
+      meridiem,
     });
   }
 
+  parseTimeTwelveHourTime(hour) {
```

Comment:
What do you think about using doc here to note what the intentions of this function are. I'm thinking some JSDoc format to describe it, the argument passed to it, and what it returns.

Reply:
+1 to inline docs since time-input can be pretty complex

Reply:
Can we move this to TimeUtils?

Reply:
Done. https://github.com/cerner/terra-core/pull/1027/commits/a3c2caa1e5915e7ba5d87e054fc695248a528bf3